server/server.go

```diff
@@ -451,6 +451,18 @@ func NewServer(opts *Options) (*Server, error) {
 				return nil, fmt.Errorf("no local account %q for remote leafnode", r.LocalAccount)
 			}
 		}
+	} else {
+		if len(opts.LeafNode.Users) != 0 {
```

Comment:
Would that be possible to move those checks in `validateLeafNode()` which checks for options?

Reply:
absolutely, I placed them here because of the non operator mode checks. Ok to move all those checks?

Reply:
That being said, I am not sure since we call this in NewServer()
```
if err := validateOptions(opts); err != nil {
    return nil, err
}
```
but before we actually process the trusted keys:
```
	// Trusted root operator keys.
	if !s.processTrustedKeys() {
		return nil, fmt.Errorf("Error processing trusted operator keys")
	}
```
would have to make sure that the validation of leafnode would still work if you move them there.